perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -911,18 +917,82 @@ public void checkFacilityContactExists(PerunSession sess, Facility facility, Str
 	}
 
 	@Override
+	public List<Service> getAssignedServices(PerunSession sess, Facility facility) throws InternalErrorException {
```

Comment:
There is no reason for this complex logic in Bl. You should always generally prevent iterating over a list of ids and getting each item by its id. Just select list of Services directly in impl layer using sql join with services table, like we do for Resource. We already have service mapper and other necessary stuff for it.